### PR TITLE
blog post: Extending LiveView with Custom Channels

### PR DESCRIPTION
**Pull request by SophieDeBenedetto**
I built on the chat app in my most recent post to build more responsive UX with the help of custom channels in LiveView.

### CODE REVIEW

File: _posts/2019-05-24-live-view-with-channels.md [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+
+## The Problem
+
+In a [recent post](https://elixirschool.com/blog/live-view-with-presence/), we built a straightforward chatting application backed by LiveView, PubSub and Presence. We implemented nearly all of the necessary features (live updates as users type in new messages, a list that keeps track of users in the chat room and who is typing!) with only 90 lines of LiveView code.
+
+But then we ran into a blocker.
+
+When new chat messages were appended to the chat window, they appeared *just* out of frame. The chat window needed to scroll down to accommodate and display the new message. This is easy enough to do with just one or two lines of JavaScript: grab the height of the chat window, and set the `scrollTop` accordingly.
+
+[GIF WILL GO HERE]
```
Comment (doomspork) [inline _posts/2019-05-24-live-view-with-channels.md]: @SophieDeBenedetto do you want to add in the gif?

File: _posts/2019-05-24-live-view-with-channels.md [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+
+In this section, we'll dive into the following portion of the process:
+
+1. User submits a new chat message, sending an event to the live view; The live view updates its state and re-renders the template
+2. The live view broadcasts the event to the other live view processes subscribing to that chat room topic which then update their own state and re-render their templates
+3. The live views send a message *to themselves*, instructing them to in turn send a message to their "associated" channel (i.e. the channel joined on the live view's socket). This ensures that the live view will finish re-rendering *before* telling the channel to push a message to the front-end.
+
+Here's a closer look at this flow:
+
+![]({% asset live-view-handle-event.png @path %})
```
Comment (doomspork) [inline _posts/2019-05-24-live-view-with-channels.md]: Should we add names to these links for accessibility?

File: _posts/2019-05-24-live-view-with-channels.md [hunk 3]
```diff
@@ -0,0 +1,499 @@
+---
+author: Sophie DeBenedetto
+author_link: https://github.com/sophiedebenedetto
+categories: post
+date: 2019-05-24
```
Comment (tajchumber) [inline _posts/2019-05-24-live-view-with-channels.md]: do you want to put the current date in?

Comment (SophieDeBenedetto) [reply _posts/2019-05-24-live-view-with-channels.md]: Good catch! I'll update this before we merge :)

**Comment by SophieDeBenedetto** [comment]
@tajchumber @doomspork addressed comments and updated date!

**Pull request merged**